src/bp/ui-studio/src/web/views/OneFlow/sidePanel/TopicEditor/ImportTopic.tsx

```diff
@@ -0,0 +1,121 @@
+import { FileInput, Icon } from '@blueprintjs/core'
+import 'bluebird-global'
+import { lang } from 'botpress/shared'
+import { FlowView } from 'common/typings'
+import _ from 'lodash'
+import React, { FC, useState } from 'react'
+import { toastFailure, toastSuccess } from '~/components/Shared/Utils'
+
+import { ElementType } from '..'
+import { ExportedFlow, ExportedTopic } from '../typings'
+import style from '../TopicList/style.scss'
+import { analyzeWorkflowFile, executeWorkflowActions, getWorkflowAction } from '../WorkflowEditor/import'
+
+import { analyzeTopicFile, detectFileType, executeTopicActions, getTopicAction, renameTopic } from './import'
+
+interface Topic {
+  name: string
+  description: string
+}
+
+interface Props {
+  onImportCompleted: () => void
+  isOpen: boolean
+  flows: FlowView[]
+  topics: Topic[]
+  selectedTopic: string
+}
+
+const ImportTopic: FC<Props> = props => {
+  const [fileContent, setFileContent] = useState<ExportedTopic | ExportedFlow>()
+
+  const readFile = (files: FileList | null) => {
+    if (!files) {
+      return
+    }
+
+    const fr = new FileReader()
+    fr.readAsArrayBuffer(files[0])
+    fr.onload = loadedEvent => {
+      try {
+        const dec = new TextDecoder('utf-8')
+        const content = JSON.parse(dec.decode(_.get(loadedEvent, 'target.result')))
+
+        analyzeImport(content, content?.name, detectFileType(content))
```

Comment:
I know that prior to this we were showing the summary of what is going to be imported and now we don't and that's fine but I think we might want to if there's a conflicting topic.

What do you think ? If not then what's the point of "analyzingImport"